src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -267,7 +267,9 @@ class ErgoWalletActor(settings: ErgoSettings,
     case ScanOnChain(newBlock) =>
       if (state.secretIsSet(settings.walletSettings.testMnemonic)) { // scan blocks only if wallet is initialized
         val nextBlockHeight = state.expectedNextBlockHeight(newBlock.height, settings.nodeSettings.isFullBlocksPruned)
-        if (nextBlockHeight == newBlock.height) {
+        // we want to scan a block either when it is its turn or when wallet is freshly created (no need to load the past)
+        val walletFreshlyCreated = state.walletPhase == WalletPhase.Created && state.getWalletHeight == 0
```

Comment:
If I remember correctly, only `state.getWalletHeight == 0` is not enough as restored wallet has also `state.getWalletHeight == 0`, that's why the `WalletPhase` introduction